direct/functionals/ssim.py

```diff
@@ -38,15 +42,15 @@ def __init__(self, win_size=7, k1=0.01, k2=0.03):
         NP = win_size**2
         self.cov_norm = NP / (NP - 1)
 
-    def forward(self, X, Y, data_range):
+    def forward(self, input_data: torch.Tensor, target_data: torch.Tensor, data_range: torch.Tensor) -> torch.Tensor:
         data_range = data_range[:, None, None, None]
```

Comment:
Are you assuming this will always be a float32? I believe the difference between torch.tensor and torch.Tensor is that the latter is fp32.

Reply:
I am not sure if this is the case but the whole software has this typing concerning Tensors, no?

Reply:
Actually this is the behaviour

```
>>> x = torch.arange(10, dtype=torch.float16)
>>> isinstance(x, torch.Tensor)
True
>>> x = torch.arange(10, dtype=torch.float32)
>>> isinstance(x, torch.Tensor)
True
```